lib/Frontend/CompilerInvocation.cpp

```diff
@@ -435,8 +435,10 @@ static bool ParseLangArgs(LangOptions &Opts, ArgList &Args,
   if (Args.hasArg(OPT_enable_experimental_additive_arithmetic_derivation))
     Opts.EnableExperimentalAdditiveArithmeticDerivedConformances = true;
 
-  if (Args.hasArg(OPT_experimental_private_intransitive_dependencies))
-    Opts.EnableExperientalPrivateIntransitiveDependencies = true;
+  Opts.EnableExperientalPrivateIntransitiveDependencies =
+      Args.hasFlag(OPT_enable_experimental_private_intransitive_dependencies,
+                   OPT_disable_experimental_private_intransitive_dependencies,
+                   /*default*/ true);
```

Comment:
Suggestion: if instead of "true" you write "Opts.EnableExperimental..." here, then the default is only defined in one place, the .h file.